### PR TITLE
Merging JS with python

### PR DESCRIPTION
**Pull request by RadekLejba**

### CODE REVIEW

File: src/presence_analyzer/utils.py [hunk 1]
```diff
@@ -102,3 +102,28 @@ def mean(items):
     Calculates arithmetic mean. Returns zero for empty lists.
     """
     return float(sum(items)) / len(items) if len(items) > 0 else 0
+
+
+def group_by_start_end(items):
+    """
+    Calculates arithmetic mean for start-end hours of each day of the week.
+    """
+    result = [
```
Comment (mjelonek92) [inline src/presence_analyzer/utils.py]: `result = [[[], []] for _ in range(7)]`

File: src/presence_analyzer/static/presence_start_end.html [hunk 2]
```diff
@@ -79,6 +79,12 @@
         google.load("visualization", "1", {packages:["corechart", "timeline"], 'language': 'pl'});
     </script>
     <script type="text/javascript">
+        function parseInterval(value) {
+                var result = new Date(1,1,1);
```
Comment (mjelonek92) [inline src/presence_analyzer/static/presence_start_end.html]: Your indentation is a bit too much here (8 spaces?)

File: src/presence_analyzer/static/presence_start_end.html [hunk 3]
```diff
@@ -96,33 +102,29 @@
...
-                        chart_div.show();
-                        loading.hide();
-                        var chart = new google.visualization.Timeline(chart_div[0]);
-                        chart.draw(data, options);
+                        $.getJSON("/api/v1/presence_start_end/"+selected_user, function(result) {
+                            $.each(result, function(index, value) {
+                                value[1] = new Date(parseInterval(value[1]));
+                                value[2] = new Date(parseInterval(value[2]));
+                            });
+                            console.log(result);
```
Comment (mjelonek92) [inline src/presence_analyzer/static/presence_start_end.html]: I don't think we need this

File: src/presence_analyzer/tests.py [hunk 4]
```diff
@@ -173,6 +173,80 @@ def test_mean(self):
             utils.mean([12, 2, 1, 5, 1]), 4.2
         )
 
+    def test_group_by_start_end(self):
+        """
+        Test second_group_by_start_end
+        """
+        testData = {
+            'testUser1': {
```
Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: Theoretically you could use `utils.get_data` cause you copied this from the previous test, so it gets kinda repetitive.

File: src/presence_analyzer/tests.py [hunk 5]
```diff
@@ -173,6 +173,80 @@ def test_mean(self):
             utils.mean([12, 2, 1, 5, 1]), 4.2
         )
 
+    def test_group_by_start_end(self):
+        """
+        Test second_group_by_start_end
```
Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: `second_group_by_start_end`?

File: src/presence_analyzer/views.py [hunk 6]
```diff
@@ -73,3 +74,23 @@ def presence_weekday_view(user_id):
...
+    """
+    data = get_data()
+    if user_id not in data:
+        log.debug('User %s not found!', user_id)
+        abort(404)
+
+    weekdays = group_by_start_end(data[user_id])
+    result = [
+        (calendar.day_abbr[weekday], mean(startEnd[0]), mean(startEnd[1]))
+        for weekday, startEnd in enumerate(weekdays)
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: Can we go with `start_end` rather than camelCase?

File: src/presence_analyzer/tests.py (hunk 4 above)
Comment (RadekLejba) [inline src/presence_analyzer/tests.py]: Well, this is not the same data acctually, its a bit different. Also utils-get_data will give alot of data into this test. As far as i read about unit test, they should be as short as possible and consume as little computing power as possible. Correct me if i am wrong please.

Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: PCs are a bit stronger nowadays and simple I/O operations are not that expensive, so you don't have to worry about computing power. Feel free to move your data to a test utility function if you're going to re-use it, so it's not actually being copied around in various cases.

**Pull request merged**